src/deparse.c

```diff
@@ -0,0 +1,220 @@
+#include <postgres.h>
+#include <catalog/namespace.h>
+
+#include <fmgr.h>
+#include <utils/rel.h>
+#include <commands/dbcommands.h> //get_database_oid(dbname)
+#include <utils/varlena.h> // for SQL text types not needed after SQL exposure is removed
+#include <utils/lsyscache.h>
+#include <utils/builtins.h>
+#include <utils/syscache.h>
+#include <access/htup_details.h>
+#include <access/heapam.h>
+#include <utils/relcache.h>
+#include <catalog/objectaccess.h> // get_catalog_object_by_oid
+#include <lib/stringinfo.h>
+#include <catalog/pg_type.h>
+#include <catalog/objectaddress.h>
+#include <catalog/pg_collation.h>
+#include <catalog/pg_constraint.h>
+#include <catalog/pg_constraint_fn.h>
+#include <access/sysattr.h> // FirstLowInvalidHeapAttributeNumber
+#include <access/genam.h> // pg_constraint scan stuff
+#include <utils/fmgroids.h>
+#include <catalog/indexing.h>
+#include <utils/ruleutils.h>
+
+#include <utils/regproc.h>
+
+void		ct_deparse_columns(StringInfo qs, Relation table_rel);
+void		ct_deparse_constraints(StringInfo qs, Relation table_rel);
+char	   *deparse_create_table(Oid reloid);
+
+PG_FUNCTION_INFO_V1(deparse_test);
```

Comment:
should probably be named deparse_create_table_test or something.

Reply:
done